src/Infrastructure/DependencyInjection.cs

```diff
@@ -0,0 +1,48 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Identity;
+
+using Codidact.Authentication.Infrastructure.Persistance;
+using Codidact.Authentication.Domain.Entities;
+
+namespace Codidact.Authentication.Infrastructure
+{
+    public static class DependencyInjection
+    {
+        public static IServiceCollection AddInfrastructure(this IServiceCollection services, IConfiguration configuration)
+        {
+            services
+                .AddDbContext<ApplicationDbContext>(options =>
+                {
+                    options.UseSqlite(configuration.GetConnectionString("Authentication"));
+                });
+
+            services
+                .AddIdentityCore<ApplicationUser>(options =>
+                {
+                    options.Password.RequireDigit = false;
+                    options.Password.RequiredLength = 8;
+                    options.Password.RequiredUniqueChars = 1;
+                    options.Password.RequireLowercase = false;
+                    options.Password.RequireNonAlphanumeric = false;
+                    options.Password.RequireUppercase = false;
+                })
+                .AddEntityFrameworkStores<ApplicationDbContext>()
+                .AddSignInManager();
+
+            services
+                .AddIdentityServer()
+                .AddInMemoryClients(configuration.GetSection("IdentityServer:Clients"))
+                .AddInMemoryIdentityResources(Config.GetIdentityResources())
+                .AddAspNetIdentity<ApplicationUser>()
+                .AddDeveloperSigningCredential();
```

Comment:
If possible add `if(enviornment.isDevelopement)` here. Better yet open an issue for this if you want to.
The problem is passing the ENV variable to the this extension method because infrastructure refuses to recognize IHostEnv no matter the nugets I installed.

Reply:
bb77ebadd11626a40ee14357be1e0b6230a50ae1

I have created #6 to do this properly, but this will do for now.